destroy.sh

```diff
@@ -106,26 +105,63 @@ do
   fi
 
   # Empty the bucket
+  echo "emptying bucket: $i"
   aws s3 rm s3://$i/ --recursive
 done
 
-echo "Removing certificate from stage"
-
 restApiName=$stage-app-api
 
 restApiId=$(aws apigateway get-rest-apis | jq -r ".[] | .[] |  select(.name==\"$restApiName\") | .id|tostring")
 
-aws apigateway update-stage \
-  --rest-api-id $restApiId \
-  --stage-name $stage \
-  --patch-operations op=replace,path=/clientCertificateId,value="" \
-  &>/dev/null
+if [ "$restApiId" != "" ]; then
+  echo "Removing certificate from stage"
+
+  aws apigateway update-stage \
+    --rest-api-id $restApiId \
+    --stage-name $stage \
+    --patch-operations op=replace,path=/clientCertificateId,value="" \
+    &>/dev/null
+
+  restApiArn="arn:aws:apigateway:us-east-1::/restapis/$restApiId/stages/$stage"
 
-echo "Removed certificate from stage"
+  echo "Disassociating web acl from api gateway"
+
+  aws wafv2 disassociate-web-acl \
+    --resource-arn $restApiArn &>/dev/null
```

Comment:
this enables the `app-api-*` stack to delete fully! currently it fails because the Web ACL is still associated